pkg/build/strategies/sti/sti.go

```diff
@@ -228,16 +228,27 @@ func (builder *STI) Build(config *api.Config) (*api.Result, error) {
 	} else {
 		glog.V(1).Infof("Running %q in %q", api.Assemble, config.Tag)
 	}
-	if err := builder.scripts.Execute(api.Assemble, config.AssembleUser, config); err != nil {
+	startTime := time.Now()
+	err := builder.scripts.Execute(api.Assemble, config.AssembleUser, config)
+	builder.result.BuildInfo.Steps = utilstatus.RecordStepInfo(builder.result.BuildInfo.Steps, api.StepAssembleBuildScripts, startTime)
+	if err != nil {
 		if err == errMissingRequirements {
 			glog.V(1).Info("Image is missing basic requirements (sh or tar), layered build will be performed")
 			return builder.layered.Build(config)
 		}
-		if _, ok := err.(s2ierr.ContainerError); ok {
-			builder.result.BuildInfo.FailureReason = utilstatus.NewFailureReason(
-				utilstatus.ReasonAssembleFailed,
-				utilstatus.ReasonMessageAssembleFailed,
-			)
+		if e, ok := err.(s2ierr.ContainerError); ok {
+			if !isMissingRequirements(e.Output) {
+				builder.result.BuildInfo.FailureReason = utilstatus.NewFailureReason(
+					utilstatus.ReasonAssembleFailed,
+					utilstatus.ReasonMessageAssembleFailed,
+				)
+				return builder.result, err
+			}
+			glog.V(1).Info("Image is missing basic requirements (sh or tar), layered build will be performed")
```

Comment:
@jeremyeder because attaining that level of instrumentation is sisyphean and going to be really hard to maintain going forward any time we add/restructure code.

there are a million things we do during builds and trying to name/define steps for all of them is going to end up as a mess, or we're going to be counting things as a single duration that aren't really related.

In practice I expect the total duration will roughly match the duration of the steps we are reporting, because we're adding timing information for any step that we expect will have a meaningful duration.  But in practice we might have a few seconds here and there that are not accounted for where we're doing some internal stuff.  If it turns out they are significantly off, that will be an indication that there is additional work we should be reporting durations for.